### PR TITLE
More tests for events

### PR DESCRIPTION
**Pull request by ekr-cfa**
I added the tests from my review.

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -38,6 +42,7 @@ type Callback = dyn FnOnce(&mut Context);
 pub struct Context {
     plan_queue: Queue<Box<Callback>>,
     callback_queue: VecDeque<Box<Callback>>,
+    event_handlers: Option<HashMap<TypeId, Box<dyn Any>>>,
```
Comment (k88hudson-cfa) [inline src/context.rs:45]: Can you improve the comments about why we need an Option

**Review by k88hudson-cfa — APPROVED** [review]
LGTM

File: src/context.rs (hunk 1 above)
Comment (jasonasher) [reply src/context.rs]: I replaced the use of Option here with destructuring.

**Review by ekr-cfa — COMMENTED** [review]
Jason's changes look good.

**Pull request merged**